### ISSUE TITLE
Use pandas in the examples to load CSV and txt files

### ISSUE THREAD

**Issue by GaelVaroquaux** [nilearn/nilearn#1140]
We should consider using pandas in the examples to load CSV files. It might make the examples more readable and also give to the users more chances of being able to load new data (ie other than what's in the Haxby example).

**Comment by lesteve** [comment]
> We should consider using pandas in the examples to load CSV files

Out of interest, would that be only for the examples for now, i.e. we wouldn't add it as a dependency of nilearn?

**Comment by GaelVaroquaux** [comment]
> Out of interest, would that be only for the examples for now, i.e. we
> wouldn't add it as a dependency of nilearn?

Indeed, that's the idea.

**Comment by AlexandreAbraham** [comment]
+1 for adding it as a dependency otherwise it would be weird for people not to be able to run examples out of the box.

**Comment by aabadie** [comment]
Maybe I can take this one. Any objection ?

**Comment by aabadie** [comment]
I like the idea but I'm wondering if it will really make things clearer for newcomers.

Example with the code taken from [examples/02_decoding/plot_haxby_anova_svm.py](https://github.com/nilearn/nilearn/blob/master/examples/02_decoding/plot_haxby_anova_svm.py):
* This the current version:
```python
# Load the behavioral data
import numpy as np
y, session = np.loadtxt(haxby_dataset.session_target[0]).astype("int").T
conditions = np.recfromtxt(haxby_dataset.conditions_target[0])['f0']
```
* Here is the `pandas` version:
```python
# Load the behavioral data
import pandas as pd
df = pd.read_csv(haxby_dataset.session_target[0], header=None,
                 names=['y', 'session'], delimiter=' ')
y = df.y.values
session = df.session.values
df = pd.read_csv(haxby_dataset.conditions_target[0], header=None,
                 names=['f0', 'f1'], delimiter=' ')
conditions = df.f0.values.astype('S12')
```

But maybe this is not the best example or I don't have a good usage of pandas.

**Comment by bthirion** [comment]
Maybe there are other places where this would be more useful. If there is none, I am rather -1 on adding the dependency.

**Comment by GaelVaroquaux** [comment]
<pre>
> df = pd.read_csv(haxby_dataset.session_target[0], header=None,
>                  names=['y', 'session'], delimiter=' ')
</pre>

You're pushing it: we could do
<pre>
df = pandas.read_csv('labels.txt', delimiter=' ')
</pre>

Later code can be modified to be more readable:
<pre>
condition_mask = df['labels'].isin(['face', 'cat'])
</pre>

That said, I agree that it does make the code much more readable.

What I have in mind with this issue is that it may be better to teach
people to use pandas to deal with CSV and columnar datat rather than
numpy.

But I am only weakly convinced. So let's put this on the side for now.

**Comment by emdupre** [comment]
I'm at the Montreal sprint and happy to pick this up!

**Comment by KamalakerDadi** [comment]
Thanks for pitching in. I am afraid this is weakly convinced to add pandas as a dependency. May be we wait others to pitch in to raise their opinion.

**Comment by GaelVaroquaux** [comment]
> Thanks for pitching in. I am afraid this is weakly convinced to add
> pandas as a dependency.

There is an emerging concensus to add it. In particular since we will
need it for nistats too.

**Comment by GaelVaroquaux** [comment]
Fixed by #1556
